torch/nn/modules/conv.py

```diff
@@ -996,3 +997,397 @@ def __init__(self, *args, **kwargs):
 # TODO: Conv2dLocal
 # TODO: Conv2dMap
 # TODO: ConvTranspose2dMap
+
+
+class _LazyConvXdMixin(LazyModuleMixin):
+
+    def reset_parameters(self) -> None:
+        if not self.has_uninitialized_params() and self.in_channels != 0:
+            super().reset_parameters()
+
+    def initialize_parameters(self, input) -> None:
+        if self.has_uninitialized_params():
+            with torch.no_grad():
+                self.in_channels = input.shape[1]
+                if self.transposed:
+                    self.weight.materialize((
+                        self.in_channels, self.out_channels // self.groups, *self.kernel_size))
+                else:
+                    self.weight.materialize((
+                        self.out_channels, self.in_channels // self.groups, *self.kernel_size))
+                self.reset_parameters()
+
+
+class LazyConv1d(_LazyConvXdMixin, Conv1d):
```

Comment:
I am not sure how much of the conv1d doc we want to copy/paste here.
Should we simply link back to the the conv1d and lazy modules docs?
In particular, I'm wondering if `UninitializedParameter` mention here might be confusing for the users.

Reply:
How about the following documentation simply? :)
```
    r"""A :class:`torch.nn.Conv1d` module with lazy initialization.

    Args:
        out_channels (int): Number of channels produced by the convolution
        kernel_size (int or tuple): Size of the convolving kernel
        stride (int or tuple, optional): Stride of the convolution. Default: 1
        padding (int or tuple, optional): Zero-padding added to both sides of
            the input. Default: 0
        padding_mode (string, optional): ``'zeros'``, ``'reflect'``,
            ``'replicate'`` or ``'circular'``. Default: ``'zeros'``
        dilation (int or tuple, optional): Spacing between kernel
            elements. Default: 1
        groups (int, optional): Number of blocked connections from input
            channels to output channels. Default: 1
        bias (bool, optional): If ``True``, adds a learnable bias to the
            output. Default: ``True``

    .. seealso:: :class:`torch.nn.Conv1d` and :class:`torch.nn.modules.lazy.LazyModuleMixin`
```

Reply:
That looks nice yes.
How does the "seealso" directive look like in the doc? Do we use that already for other functions?

I would add a bit more precision at the beginning to mention what is actually "lazy" here: "with lazy initialization of the `in_channels` argument of the Conv1d that is inferred from the `input.size(1)`"

Reply:
For example, [torch.remainder](https://pytorch.org/docs/stable/generated/torch.remainder.html) has a `.. seealso::` section.
PyTorch codebase seems to have both notations. Which are you preferred?
https://github.com/pytorch/pytorch/search?q=%22..+seealso%3A%3A%22
https://github.com/pytorch/pytorch/search?q=%22see+also%22

Reply:
I think the `.. seealso:` is good. I was mostly curious how it looked like. Thanks!

Reply:
I think a small update of this paragraph is still needed to remove the mention to `UninitializedParameter` and explain what is infered and from where.

Reply:
Fixed in eaeb5de. Does it reflect your intentions correctly?

Reply:
I would just add the "with lazy initialization of the in_channels argument of the Conv1d that is inferred from the input.size(1)" just to make it clear to the user what we actually lazily initialize :)

Reply:
Sorry, I totally forgot it. Now I updated!
https://github.com/pytorch/pytorch/compare/eaeb5de9de35c07558d077e0293eafb576a7589a..7c0382c520b4fb1382bae247f69613d66b4f21be